nautobot/ipam/views.py

```diff
@@ -779,11 +796,20 @@ class IPAddressAssignView(generic.ObjectView):
 
     queryset = IPAddress.objects.all()
 
+    def _get_interface_type_and_model(self, request):
+        interface_model = Interface if "interface" in request.GET else VMInterface
+        interface_id = request.GET.get("interface") or request.GET.get("vminterface")
+        return interface_id, interface_model
```

Comment:
Since you have this method, can you please replace the calls on lines 715-716 and lines 767-768 with this?